## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 509882`
Commit: `411f65783748c48f88eb39ddc7d3b0d89ca6f106`

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fosrl-newt</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fosrl-olm</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>fosrl-newt</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fosrl-olm</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>fosrl-newt</summary>
<pre>    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:53255-&gt;[::]:57119: sendto: no route to host
--- FAIL: TestNetstackRouting (0.00s)
    shared_bind_test.go:472: Send failed: write udp [::]:60724-&gt;[::]:63630: sendto: no route to host
--- FAIL: TestSocketRouting (0.00s)
    shared_bind_test.go:535: Send failed: write udp [::]:63382-&gt;[::]:57068: sendto: no route to host
FAIL
FAIL	github.com/fosrl/newt/bind	0.484s
FAIL</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>fosrl-newt</summary>
<pre>    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:309: Hole punch WriteToUDP failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
    shared_bind_test.go:296: WireGuard Send failed: write udp [::]:64825-&gt;[::]:49720: sendto: no route to host
--- FAIL: TestNetstackRouting (0.00s)
    shared_bind_test.go:472: Send failed: write udp [::]:63994-&gt;[::]:58194: sendto: no route to host
--- FAIL: TestSocketRouting (0.00s)
    shared_bind_test.go:535: Send failed: write udp [::]:54521-&gt;[::]:53406: sendto: no route to host
FAIL
FAIL	github.com/fosrl/newt/bind	0.271s
FAIL</pre>
</details>
</details>

--- a/pkgs/by-name/fo/fosrl-newt/package.nix
+++ b/pkgs/by-name/fo/fosrl-newt/package.nix
@@ -34,6 +34,8 @@ buildGoModule (finalAttrs: {
 
   passthru.updateScript = nix-update-script { };
 
+  __structuredAttrs = true;
+
   meta = {
     # Networking failures in tests, even with __darwinAllowLocalNetworking on
     # and sandbox disabled.

--- a/pkgs/by-name/fo/fosrl-newt/package.nix
+++ b/pkgs/by-name/fo/fosrl-newt/package.nix
@@ -4,6 +4,7 @@
   fetchFromGitHub,
   versionCheckHook,
   nix-update-script,
+  stdenv,
 }:
 
 buildGoModule (finalAttrs: {
@@ -34,6 +35,10 @@ buildGoModule (finalAttrs: {
   passthru.updateScript = nix-update-script { };
 
   meta = {
+    # Networking failures in tests, even with __darwinAllowLocalNetworking on
+    # and sandbox disabled.
+    # Unclear as of 2026-04-24 whether the program works if tests are disabled.
+    broken = stdenv.hostPlatform.isDarwin;
     description = "Tunneling client for Pangolin";
     homepage = "https://github.com/fosrl/newt";
     changelog = "https://github.com/fosrl/newt/releases/tag/${finalAttrs.src.tag}";

--- a/pkgs/by-name/fo/fosrl-olm/package.nix
+++ b/pkgs/by-name/fo/fosrl-olm/package.nix
@@ -24,6 +24,8 @@ buildGoModule (finalAttrs: {
 
   doInstallCheck = true;
 
+  __structuredAttrs = true;
+
   meta = {
     description = "Tunneling client for Pangolin";
     homepage = "https://github.com/fosrl/olm";

